test/Tester/CancellationTests/CancellationTokenPassingTests.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans;
+using Orleans.TestingHost;
+using UnitTests.GrainInterfaces;
+using Xunit;
+using UnitTests.Tester;
+
+namespace UnitTests.MembershipTests
+{
+    public class CancellationTokenPassingTests : HostedTestClusterPerTest
+    {
+        private double[] cancellationDelaysInMS = { 0, 0.1, 0.5, 1, 20, 500, 1000 };
+
+        public override TestingSiloHost CreateSiloHost()
+        {
+            return new TestingSiloHost(new TestingSiloOptions
+            {
+                StartFreshOrleans = true,
+                StartPrimary = true,
+                StartSecondary = true,
+                AdjustConfig = config =>
+                {
+                    config.Globals.DefaultPlacementStrategy = "ActivationCountBasedPlacement";
+                }
+            });
+        }
+
```

Comment:
What would happen if `CancellationToken`s were linked together with [CreateLinkedTokenSource ](https://msdn.microsoft.com/en-us/library/ee191553(v=vs.110).aspx), say, in a fan-out scenario? (cf. http://blogs.msdn.com/b/pfxteam/archive/2009/05/22/9635790.aspx).